### PR TITLE
Postcss

### PR DESCRIPTION
**Pull request by devinus**

### CODE REVIEW

**Comment by motdotla** [comment]
woot. :+1:

I havent pulled down locally but tests are passing, and code is additionally cleaner. Looks good.

File: lib/main.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var _ = require('lodash');
-var css = require('css');
+var postcss = require('postcss');
```
Comment (motdotla) [inline lib/main.js]: Looks like this and its dependency in package.json can be removed

**Comment by devinus** [comment]
Don't merge yet. I still want to do a few more improvements. I think there is a bug when using a class name that matches the element name.

E.g. `label .label` will become `xy .xy`.

File: lib/main.js (hunk 1 above)
Comment (motdotla) [inline lib/main.js]: Ignore that i see line 179 does use it. I see, combo of postcss and csswhat

Comment (devinus) [inline lib/main.js]: Yes, `postcss` is the same CSS parser that `autoprefixer` uses, and `css-what` is the same selector parsing engine that `cheerio` uses.

Comment (devinus) [inline lib/main.js]: With bigger communities they should have less bugs and handle more edge-cases, which is crucial for us because of our mmmmega hax.

**Comment by motdotla** [comment]
Want me to pick this one up @devinus? I have open bandwidth.

**Comment by devinus** [comment]
@motdotla Selfishly, not really. I'd love to be able to hack on this on my flight tomorrow. It's not needed for TS since it changes no functionality other than fixing the `@media{` bug so far.

**Comment by motdotla** [comment]
Sounds good :)

**Comment by devinus** [comment]
@motdotla Ready for review.

File: lib/main.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+
 var cheerio = require('cheerio');
+var postcss = require('postcss');
+var cssWhat = require('css-what');
 var Hashids = require('hashids');
 
-var ALPHABET = 'abcdefghijklmnopqrstuvwxyz';
+var stringify = require('./css-what/stringify');
+
+var ALPHABET = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
```
Comment (motdotla) [inline lib/main.js]: Does html read  `"somID"` differently than `"someid"`. I think it's case insensitive - so having the upper case could be dangerous here?

File: lib/main.js [hunk 3]
```diff
@@ -11,217 +15,138 @@ function HtmlUglify(config) {
...
-  existingPointer = lookups[lookupKey];
+  var lookupKey = [type, value].join(LOOKUP_DELIMITER);
+  var existingPointer = lookups[lookupKey];
 
+  var pointer;
   if (existingPointer) {
     pointer = existingPointer;
   } else {
-    counter = _.keys(lookups).length;
+    var counter = Object.keys(lookups).length;
```
Comment (motdotla) [inline lib/main.js:31]: :+1:

File: lib/main.js (hunk 2 above)
Comment (devinus) [inline lib/main.js]: You know what, only in quirks mode are they case-insensitive. And knowing email clients, I think it's best to just assume anything can go wrong. Reverting.

**Comment by motdotla** [comment]
Nice cleanup. Tighter. 

My only thought is the one comment I left concerning the available alphabet.

**Comment by motdotla** [comment]
Merge away :+1:

**Pull request merged**